pkg/webhook/v1beta1/pod/inject_webhook_test.go

```diff
@@ -213,23 +208,19 @@ func TestWrapWorkerContainer(t *testing.T) {
 					},
 				},
 			},
-			Err:             false,
-			TestDescription: "Container with early stopping command",
 		},
 	}
 
-	for _, c := range testCases {
-		err := wrapWorkerContainer(c.Trial, c.Pod, c.Trial.Namespace, c.MetricsFile, c.PathKind)
-		if c.Err && err == nil {
-			t.Errorf("Case %s failed. Expected error, got nil", c.TestDescription)
-		} else if !c.Err {
-			if err != nil {
-				t.Errorf("Case %s failed. Expected nil, got error: %v", c.TestDescription, err)
-			} else if !equality.Semantic.DeepEqual(c.Pod.Spec.Containers, c.ExpectedPod.Spec.Containers) {
-				t.Errorf("Case %s failed. Expected pod: %v, got: %v",
-					c.TestDescription, c.ExpectedPod.Spec.Containers, c.Pod.Spec.Containers)
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			err := wrapWorkerContainer(tc.Trial, tc.Pod, tc.Trial.Namespace, tc.MetricsFile, tc.PathKind)
+			if diff := cmp.Diff(tc.WantError, err, cmpopts.EquateErrors()); len(diff) != 0 {
+				t.Errorf("Unexpected error from wrapWorkerContainer (-want,+got):\n%s", diff)
 			}
-		}
+			if err == nil && !equality.Semantic.DeepEqual(tc.Pod.Spec.Containers, tc.WantPod.Spec.Containers) {
```

Comment:
Could you apply the cmp approach here as well?

Reply:
This has been fixed in this [commit](https://github.com/kubeflow/katib/pull/2289/commits/f2b0ca4163afc29593144ee9d69612b7f0d00d6e).